### ISSUE TITLE
Lint/ConstantResolution and private_constant

### ISSUE THREAD

**Issue by scottohara** [rubocop/rubocop#9173]
I have enabled the `Lint/ConstantResolution` cop and have been liberally adding `::` prefixes to all of the bare constants in my code base to satisfy the offenses.

I've encountered an issue with `private_constant` though, and I'm not entirely sure how to resolve.

(disclaimer: I'm a bit rusty, as it has been a while since I've worked with Ruby, so apologies in advance if there's a simple explanation).

--------

## Expected behavior

Either no `Lint/ConstantResolution` offenses when a private constant is referenced; or for the error message to provide more detail on how to resolve (e.g. provide an example)

## Actual behavior

It it unclear (at least to me, a novice ruby programmer) how to satisfy the cop when using private constants.

## Steps to reproduce the problem

In the example below, a module `Foo` defines a private constant `BAR`, which is later referenced in `Foo::Baz#do_something`

```ruby
module Foo
  BAR = 1
  private_constant :BAR

  class Baz
    def do_something
      BAR      # <-- offense here
    end
  end
end
```

When `Lint/ConstantResolution` is enabled, an offense at line 7 is triggered:

``` 
Fully qualify this constant to avoid possibly ambiguous resolution.
```

Changing the constant reference to `::BAR` in this instance won't work because it's private, and causes a `NameError (uninitialized constant BAR)` error at runtime.

Equally, changing to `Foo::BAR` or `::Foo::BAR` results in a `NameError (private constant Foo::BAR referenced)` error at runtime.

What would be the fully qualified name for a private constant that both satisfies the cop, but still allows code in the module to reference the constant?

I'm aware that `Lint/ConstantResolution` has `Only: []` and `Ignore: []` options to allow/block specific constants, but I'm interested if there's a way to for this cop to handle private constants.

## RuboCop version

```
❯ bundle exec rubocop -V
1.5.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.7.2 x86_64-darwin18)
  - rubocop-performance 1.9.1
  - rubocop-rails 2.8.1
  - rubocop-rspec 2.0.1
```

**Comment by dvandersluis** [comment]
This is a fair point that I don't know if there's an easy fix for. RuboCop doesn't resolve identifiers, so the code flagging this offense has no idea that `BAR` is private (it could even be defined, or made private, in a completely separate file, and RuboCop generally works on a single file at a time).

Unfortunately I think either using `# rubocop:disable Lint/ConstantResolution` or adding `BAR` to the `Ignore` configuration is probably your best bets here.

**Comment by jfelchner** [comment]
I'd like to go for a third option here.  I see much less of an issue with `UPPER_SNAKE_CASE_CONSTANTS` being "mispathed" than `CamelCaseClassOrModuleConstants`.  I think it might be a good middle ground to allow `Ignore` to take a regex and just apply `\A[A-Z_]+\z` to it.

**Comment by bbatsov** [comment]
Closing as stale — `Lint/ConstantResolution` is disabled by default and this edge case with `private_constant` hasn't attracted engagement in 5+ years. If this is still relevant, please reopen.
